File: C++/682. Baseball Game.cpp
class Solution {
public:
    int calPoints(vector<string>& a) {
        vector<int>d;
        int x,y,i;
        for(i=0;i<a.size();i++)
        {
            if(a[i]=="+")
            {
                x=d[d.size()-1];
                y=d[d.size()-2];
                d.push_back(x+y);

            }
            else if(a[i]=="D")
            {
                x=d[d.size()-1];
                d.push_back(x*2);
            }
            else if(a[i]=="C")
            {
                d.pop_back();
            }
            else
            {
                x=stoi(a[i]);
                d.push_back(x);
            }
        }
    return(accumulate(d.begin(), d.end(), 0));
    }
};

File: JAVA/682. Baseball Game.java
import java.util.Stack;

class Solution {
    public int calPoints(String[] o) {
        Stack<Integer> d = new Stack<>();
        int s = 0;
        
        for (int i = 0; i < o.length; i++) {
            if (o[i].equals("+")) {
                int x = d.get(d.size() - 2) + d.get(d.size() - 1);
                d.push(x);
            } else if (o[i].equals("D")) {
                int x = d.peek();
                d.push(x * 2);
            } else if (o[i].equals("C")) {
                d.pop();
            } else {
                int x = Integer.parseInt(o[i]);
                d.push(x);
            }
        }
        
        for (int i : d) {
            s += i;
        }
        
        return s;
    }
}
